### PR TITLE
COMPASS-4401 Add check for readonly before drop event

### PR DESCRIPTION
**Pull request by aherlihy**
COMPASS-4401

### Checklist
- [ ] New tests and/or benchmarks are included
- [ ] Documentation is changed or added

## Types of changes

- [x] Patch (non-breaking change which fixes an issue)
- [ ] Minor (non-breaking change which adds functionality)
- [ ] Major (fix or feature that would cause existing functionality to change)

### CODE REVIEW

**Review by mcasimir — APPROVED** [review]
Cool!

File: src/components/sidebar-collection/sidebar-collection.jsx [hunk 1]
```diff
@@ -31,9 +31,11 @@ class SidebarCollection extends PureComponent {
...
   onDrop = () => {
-    const databaseName = this.props.database;
-    const collectionName = this.getCollectionName();
-    this.props.globalAppRegistryEmit('open-drop-collection', databaseName, collectionName);
+    if (!this.isNotWritable()) {
+      const databaseName = this.props.database;
+      const collectionName = this.getCollectionName();
+      this.props.globalAppRegistryEmit('open-drop-collection', databaseName, collectionName);
+    }
   }
```
Comment (addaleax) [inline src/components/sidebar-collection/sidebar-collection.jsx:39]: Should we also do this for `onImportData()` directly below?

**Review by addaleax — APPROVED** [review]

File: src/components/sidebar-collection/sidebar-collection.jsx [hunk 2]
```diff
@@ -31,9 +31,11 @@ class SidebarCollection extends PureComponent {
    * Handle drop collection.
    */
   onDrop = () => {
-    const databaseName = this.props.database;
-    const collectionName = this.getCollectionName();
-    this.props.globalAppRegistryEmit('open-drop-collection', databaseName, collectionName);
+    if (!this.isNotWritable()) {
```
Comment (mcasimir) [inline src/components/sidebar-collection/sidebar-collection.jsx:34]: Can it be that: 

```
if(this.isNotWritable()) {
return;
}
```

reads better than the not not..?

File: src/components/sidebar-collection/sidebar-collection.jsx (hunk 1 above)
Comment (aherlihy) [reply src/components/sidebar-collection/sidebar-collection.jsx:39]: onImportData is emitted and picked up by a separate plugin, which already has the check. It's not ideal but it's not currently broken so I wasn't sure if it made sense to add for symmetry even if it wouldn't matter?

File: src/components/sidebar-collection/sidebar-collection.jsx (hunk 2 above)
Comment (aherlihy) [reply src/components/sidebar-collection/sidebar-collection.jsx:34]: Eh, I'm not sure it really matters

File: src/components/sidebar-collection/sidebar-collection.jsx (hunk 1 above)
Comment (addaleax) [reply src/components/sidebar-collection/sidebar-collection.jsx:39]: No, totally fine, I was just curious :+1:

**Pull request merged**